Automattic/wp-calypso
# Editor: Add a deprecation dialog

#### Changes proposed in this Pull Request

This change adds a popup deprecation dialog to the Calypso editor, warning that it will soon be replaced by the block editor.

#### Testing instructions

* Visit `/post/mysite.wordpress.com`
* (If you're redirected to the block editor, open the `...` menu and select "Switch to Classic Editor".)
* Confirm that the dialog pops up.
* Confirm that it can be dismissed.
* Confirm that it doesn't display again after being dismissed.
* Confirm that clicking the "Try it out" button switches you to the block editor.
* After being switched to the block editor, confirm that visiting `/post/mysite.wordpress.com` will continue redirecting you to the block editor.

Fixes #41084.
